analytical_engine/core/utils/trivial_tensor.h

```diff
@@ -69,5 +70,40 @@ struct trivial_tensor_t {
   std::vector<size_t> shape_;
   T* data_;
 };
+
+template <>
+struct trivial_tensor_t<std::string> {
+ public:
+  trivial_tensor_t() : size_(0) {}
+
+  ~trivial_tensor_t() = default;
+
+  std::vector<std::string>& data() { return data_; }
+
+  const std::vector<std::string>& data() const { return data_; }
+
+  void fill(const std::string& value) { data_.push_back(value); }
```

Comment:
`fill` usually means **filling all cells with the given value** (see https://en.cppreference.com/w/cpp/algorithm/fill).

I think here you are meaning something like `append` or `push_back`.

Reply:
fix